src/app/shared/components/login-form/login-form.component.html

```diff
@@ -1,27 +1,53 @@
 <h2 class="text-align-center">Login</h2>
 <div class="app-plate">
-  <form #loginForm="ngForm">
+  <form #loginForm="ngForm" (ngSubmit)="onSubmit()">
     <div class="login-form">
       <div>
         <div class="form__control">
           <label for="email">Email</label>
           <!--Use the name `email` for the form control.-->
-          <input type="text" id="email" placeholder="Input text"/>
-          <span id="emailErrorMessage" class="text-danger">Email is required.</span>
+          <input type="text" id="email" name="email" placeholder="Input text" autocomplete="email"
+                 emailValidator required [(ngModel)]="email" #emailControl="ngModel"
+                 [ngClass]="{invalid:emailControl.invalid && (emailControl.touched || loginForm.submitted),
+                             valid:emailControl.valid && (emailControl.touched || loginForm.submitted),
+                             init:!emailControl.touched && !loginForm.submitted}"
+          />
+          <ng-container *ngIf="emailControl.touched || loginForm.submitted">
+            <ng-container *ngIf="emailControl.errors?.['required']">
+              <span id="emailErrorMessage" class="text-danger">Email is required.</span>
+            </ng-container>
+            <ng-container *ngIf="emailControl.errors?.['invalidEmail']">
+              <span id="emailErrorMessage" class="text-danger">Email is invalid.</span>
+            </ng-container>
+          </ng-container>
         </div>
+
         <div class="form__control">
           <label for="password">Password</label>
           <!--Use the name `password` for the form control.-->
-          <input type="password" id="password" placeholder="Input text"/>
-          <span id="passwordErrorMessage" class="text-danger">Password is required.</span>
+          <div class="password-container">
+            <input type="{{hideableRef.visible ? 'text' : 'password'}}" id="password" name="password" placeholder="Input text" 
```

Comment:
I think you can move the type attribute assignment to the directive. More on the directive.ts

Reply:
Hmm yes. Nice opportunity. Applied